Marlin/Marlin_main.cpp

```diff
@@ -628,7 +628,11 @@ float cartes[XYZ] = { 0 };
 static bool send_ok[BUFSIZE];
 
 #if HAS_SERVOS
-  Servo servo[NUM_SERVOS];
+  #if defined(__MK64FX512__) || defined(__MK65FX1M0__)
```

Comment:
The fundamental idea of the HAL is that all platform specific code is confined to the relevant HAL_xxx folder. The main code should not contain any references such as __MK64FX512__. Putting in platform specific references for every platform will make the code unmaintainable, and a lot harder to port to new targets.

This comment applies in several other places stepper.cpp, temperature.cpp